library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java

```diff
@@ -0,0 +1,28 @@
+package com.novoda.downloadmanager.lib;
+
+class RestartTimeCreator {
+
+    private final RandomNumberGenerator randomNumberGenerator;
+
+    public RestartTimeCreator(RandomNumberGenerator randomNumberGenerator) {
+        this.randomNumberGenerator = randomNumberGenerator;
+    }
+
+    /**
+     * Returns the time when a download should be restarted.
+     */
+    public long restartTime(FileDownloadInfo fileDownloadInfo, long now) {
```

Comment:
could use the `RealSystemFacade` to get the current time https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/RealSystemFacade.java